core/domain/skill_services.py

```diff
@@ -0,0 +1,331 @@
+    from the datastore.
+
+    Args:
+        skill_model: SkillModel. The skill model loaded from the
+            datastore.
+        run_conversion: bool. If true, the the skill's schema version will
+            be checked against the current schema version. If they do not match,
+            the skill will be automatically updated to the latest schema
+            version.
+
+    Returns:
+        skill. A Skill domain object corresponding to the given
+        skill model.
+    """
+
+    # Ensure the original skill model does not get altered.
+    versioned_skill_contents = {
+        'schema_version': skill_model.skill_contents_schema_version,
+        'skill_contents': copy.deepcopy(skill_model.skill_contents)
+    }
+
+    versioned_misconceptions = {
+        'schema_version': skill_model.misconceptions_schema_version,
+        'misconceptions': copy.deepcopy(skill_model.misconceptions)
+    }
+
+    # Migrate the skill if it is not using the latest schema version.
+    if (run_conversion and skill_model.skill_contents_schema_version !=
+            feconf.CURRENT_SKILL_CONTENTS_SCHEMA_VERSION):
+        _migrate_skill_contents_to_latest_schema(versioned_skill_contents)
+
+    if (run_conversion and skill_model.misconceptions_schema_version !=
+            feconf.CURRENT_MISCONCEPTIONS_SCHEMA_VERSION):
+        _migrate_misconceptions_to_latest_schema(versioned_misconceptions)
+
+    return skill_domain.Skill(
+        skill_model.id, skill_model.description,
+        [
+            skill_domain.Misconception.from_dict(misconception)
+            for misconception in versioned_misconceptions['misconceptions']
+        ], skill_domain.SkillContents.from_dict(
+            versioned_skill_contents['skill_contents']),
+        versioned_misconceptions['schema_version'],
+        versioned_skill_contents['schema_version'],
+        skill_model.language_code,
+        skill_model.version, skill_model.created_on,
+        skill_model.last_updated)
+
+
+def get_skill_summary_from_model(skill_summary_model):
+    """Returns a domain object for an Oppia skill summary given a
+    skill summary model.
+
+    Args:
+        skill_summary_model: SkillSummaryModel.
+
+    Returns:
+        SkillSummary.
+    """
+    return skill_domain.SkillSummary(
+        skill_summary_model.id, skill_summary_model.description,
+        skill_summary_model.language_code,
+        skill_summary_model.version,
+        skill_summary_model.misconception_count,
+        skill_summary_model.skill_model_created_on,
+        skill_summary_model.skill_model_last_updated
+    )
+
+
+def get_skill_by_id(skill_id, strict=True, version=None):
+    """Returns a domain object representing a skill.
+
+    Args:
+        skill_id: str. ID of the skill.
+        strict: bool. Whether to fail noisily if no skill with the given
+            id exists in the datastore.
+        version: str or None. The version number of the skill to be
```

Comment:
str not int? (Please check.)

Though actually for skills I _think_ we will always want the latest one. But I'm fine with what you have as well.

Reply:
Done, changed here and in collection_services to int.